src/astarte-charts/react/MapChart/index.tsx

```diff
@@ -0,0 +1,88 @@
+/*
+   This file is part of Astarte.
+
+   Copyright 2020 Ispirata Srl
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+import React, { useMemo } from 'react';
+import _ from 'lodash';
+import type { AstarteChartProvider } from 'astarte-charts';
+
+import { useChartProviders } from '../hooks';
+
+interface AstarteMapChartProps {
+  providers: AstarteChartProvider[];
+  height?: string | number;
+  width?: string | number;
+  zoom?: 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9 | 10 | 11 | 12 | 13 | 14 | 15 | 16 | 17 | 18 | 19 | 20;
+  showSatellite?: boolean;
+}
+
+export const AstarteMapChart = ({
+  providers,
+  height,
+  width,
+  zoom = 15,
+  showSatellite = false,
+}: AstarteMapChartProps): React.ReactElement => {
+  const deviceValues = useChartProviders(providers);
+
+  const currentPosition = useMemo(() => {
+    if (deviceValues == null) {
+      return null;
+    }
+    const orderedValues = _.orderBy(deviceValues, 'timestamp', 'desc');
+    const latitude = Number(_.find(orderedValues, (value) => value.label === 'latitude')?.value);
+    const longitude = Number(_.find(orderedValues, (value) => value.label === 'longitude')?.value);
+    return { latitude, longitude };
+  }, [deviceValues]);
+
+  return (
+    <div
+      style={{
+        width,
```

Comment:
I am not setting a specific height and width since I am relying on passed props. This makes sure a good aspect ratio (square) is still present even when a height is not specified, otherwise the Map would collapse.
How would you improve it?

Reply:
It's the parent that defines how it's children are sized and spaced.
Here you can use a HOC (higher order component) where you accept custom html attributes and merge them with your own, or return the iframe directly and let the developer handle it depending on how he sees fit.